x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/esnative/PasswordHashPromotionIntegTests.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.esnative;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.elasticsearch.cluster.health.ClusterHealthStatus;
+import org.elasticsearch.cluster.health.ClusterIndexHealth;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.LicenseService;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.test.SecuritySingleNodeTestCase;
+import org.elasticsearch.xpack.core.security.user.ElasticUser;
+
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken.basicAuthHeaderValue;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class PasswordHashPromotionIntegTests extends SecuritySingleNodeTestCase {
```

Comment:
Added one where the index exists, but the elasticuser doesn't. 
`securityIndexExistsAuthenticate` is where elasticuser already exists

Reply:
>  securityIndexExistsAuthenticate is where elasticuser already exists

But this wasn't a test and we removed it. I was looking more for a test where we index the document for the elastic user manually and then we try to authenticate with the autoconfigured password hash and it fails. If you think this is easy/quick enough to add you can do, otherwise leave a TODO and we'll follow up, let's not block this.